rewrite-java-11/src/main/java/org/openrewrite/java/Java11ParserVisitor.java

```diff
@@ -1729,6 +1730,16 @@ private JavaType type(@Nullable com.sun.tools.javac.code.Type type, List<Symbol>
                     if (sym.owner instanceof Symbol.ClassSymbol) {
                         owner = TypeUtils.asFullyQualified(type(sym.owner.type, stackWithSym));
                     }
+
+
+                    // Add annotation information to JavaType.Class
+                    List<JavaType.Class> annotations = new ArrayList<>();
+                    if(sym.getMetadata() != null && sym.getMetadata().getDeclarationAttributes() != null ) {
+                        //  TODO: create type by recursive call (see 1700) ?!
+                        sym.getMetadata().getDeclarationAttributes().stream()
+                                .forEach(a -> annotations.add(JavaType.Class.build(a.type.toString())));
+                    }
```

Comment:
Probably ok, but any code in the parser visitor (and JavaType) is on the "hot path". I would consider not using streams here and use a "for each" instead.